src/utils/files.ts

```diff
@@ -1,3 +1,34 @@
+import { Result, err, ok } from "neverthrow"
+
+import EmptyStringError from "@root/errors/EmptyStringError"
+import { PathInfo } from "@root/types/pages"
+
 export const getFileExt = (fileName: string): string =>
   // NOTE: will never be `undefined` as `fileName` is guaranteed to be a string
   fileName.split(".").pop()!
+
+export const extractPathInfo = (
+  pageName: string
+): Result<PathInfo, EmptyStringError> => {
+  if (!pageName) {
+    return err(new EmptyStringError())
+  }
+
+  const fullPath = pageName.split("/")
+  // NOTE: Name is guaranteed to exist
+  // as this method only accepts a string
+  // and we've validated that the string is not empty
+  const name = fullPath.pop()!
+
+  if (fullPath.length === 0) {
+    return ok({
+      name,
+      path: err([]),
```

Comment:
I am confused here, why are we wrapping an err around an ok here ah?

Reply:
uhm just checking in, we expect the caller to meaningfully use the `name` then? if yes, feel free to resolve this.

Reply:
yea - if it's rooted at `/` (homepage is `/index.md` for eg), then the `name` will be `index.md` but path is empty (`err`).

hence, `name` is always semantically meaningful but path depends